great_expectations/core/expectation_validation_result.py

```diff
@@ -96,7 +96,14 @@ def __eq__(self, other):
                     # numpy) consequently, series' comparison can persist. Wrapping in all() ensures comparison is
                     # handled appropriately.
                     (self.result is None and other.result is None)
-                    or (all(self.result) == all(other.result)),
+                    or all(
+                        [
+                            self.result[k] == other.result[k]
+                            for k in set(self.result.keys()).intersection(
+                                other.result.keys()
+                            )
+                        ]
+                    ),
```

Comment:
<3

Reply:
@Shinnnyshinshin I updated this section further -- to make it more robust and easier to read.